vizro-core/src/vizro/_vizro.py

```diff
@@ -37,25 +37,26 @@ def __init__(self, **kwargs):
         # Include Vizro assets (in the static folder) as external scripts and stylesheets. We extend self.dash.config
         # objects so the user can specify additional external_scripts and external_stylesheets via kwargs.
         vizro_assets_folder = Path(__file__).with_name("static")
-        static_url_path = self.dash.config.requests_pathname_prefix + STATIC_URL_PREFIX
-        vizro_css = self._get_external_assets(static_url_path, vizro_assets_folder, "css")
+        requests_pathname_prefix = self.dash.config.requests_pathname_prefix
+        vizro_css = [requests_pathname_prefix + path for path in self._get_external_assets(vizro_assets_folder, "css")]
         vizro_js = [
-            {"src": path, "type": "module"}
-            for path in self._get_external_assets(static_url_path, vizro_assets_folder, "js")
+            {"src": requests_pathname_prefix + path, "type": "module"}
+            for path in self._get_external_assets(vizro_assets_folder, "js")
         ]
         self.dash.config.external_stylesheets.extend(vizro_css)
         self.dash.config.external_scripts.extend(vizro_js)
 
         # Serve all assets (including files other than css and js) that live in vizro_assets_folder at the
         # route /vizro. Based on code in Dash.init_app that serves assets_folder. This respects the case that the
         # dashboard is not hosted at the root of the server, e.g. http://www.example.com/dashboard/vizro.
-        blueprint_prefix = self.dash.config.routes_pathname_prefix.replace("/", "_").replace(".", "_")
+        routes_pathname_prefix = self.dash.config.routes_pathname_prefix
+        blueprint_prefix = routes_pathname_prefix.replace("/", "_").replace(".", "_")
         self.dash.server.register_blueprint(
             flask.Blueprint(
                 f"{blueprint_prefix}vizro_assets",
                 self.dash.config.name,
                 static_folder=vizro_assets_folder,
-                static_url_path=static_url_path,
+                static_url_path=routes_pathname_prefix + STATIC_URL_PREFIX,
```

Comment:
This is the only line that's actually changed functionally because it now uses `routes_pathname_prefix` and not `requests_pathname_prefix`.

The other changes are just refactoring now that the `static_url_prefix` is no longer required.